hathor/indexes/addresses_rocksdb_index.py

```diff
@@ -0,0 +1,174 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Iterable, List, Optional, Set, Tuple
+
+from structlog import get_logger
+
+from hathor.indexes.addresses_index import AddressesIndex
+from hathor.pubsub import HathorEvents
+from hathor.transaction import BaseTransaction
+from hathor.transaction.scripts import parse_address_script
+
+if TYPE_CHECKING:  # pragma: no cover
+    import rocksdb
+
+    from hathor.pubsub import EventArguments, PubSubManager
+    from hathor.transaction import TxOutput
+
+logger = get_logger()
+
+_CF_NAME_ADDRESS_INDEX = b'address-index'
+
+
+class AddressesRocksDBIndex(AddressesIndex):
+    """ Index of inputs/outputs by address.
+
+    This index uses rocksdb and the following key format:
+
+        key = [address][tx.timestamp][tx.hash]
+              |--34b--||--4 bytes---||--32b--|
+
+    It works nicely because rocksdb uses a tree sorted by key under the hoods.
+
+    The timestamp must be serialized in big-endian, so ts1 > ts2 implies that bytes(ts1) > bytes(ts2),
+    hence the transactions are sorted by timestamp.
+    """
+    def __init__(self, db: 'rocksdb.DB', *, cf_name: Optional[bytes] = None,
+                 pubsub: Optional['PubSubManager'] = None) -> None:
+        self._db = db
+        self._cf_name = cf_name or _CF_NAME_ADDRESS_INDEX
+        self._cf = self._db.get_column_family(self._cf_name)
+        self.pubsub = pubsub
+        if self.pubsub:
+            self.subscribe_pubsub_events()
+
+    def _to_key(self, address: str, tx: Optional[BaseTransaction] = None) -> bytes:
+        import struct
+        assert len(address) == 34
+        key = address.encode('ascii')
+        if tx:
+            assert tx.hash is not None
+            assert len(tx.hash) == 32
+            key += struct.pack('>I', tx.timestamp) + tx.hash
+            assert len(key) == 34 + 4 + 32
+        return key
+
+    def _from_key(self, key: bytes) -> Tuple[str, int, bytes]:
+        import struct
+        assert len(key) == 34 + 4 + 32
+        address = key[:34].decode('ascii')
+        timestamp: int
+        (timestamp,) = struct.unpack('>I', key[34:38])
+        tx_hash = key[38:]
+        assert len(address) == 34
+        assert len(tx_hash) == 32
+        return address, timestamp, tx_hash
+
+    def subscribe_pubsub_events(self) -> None:
+        """ Subscribe wallet index to receive voided/winner tx pubsub events
+        """
+        assert self.pubsub is not None
+        # Subscribe to voided/winner events
+        events = [HathorEvents.STORAGE_TX_VOIDED, HathorEvents.STORAGE_TX_WINNER]
+        for event in events:
+            self.pubsub.subscribe(event, self.handle_tx_event)
+
+    def _get_addresses(self, tx: BaseTransaction) -> Set[str]:
```

Comment:
This method is exactly the same as in the `AddressesMemoryIndex`. Should we refactor this to have a helper method for it?

Reply:
Makes sense. I'll move this to `BaseTransaction.get_related_addresses`, what do you think?

Reply:
Sounds good